cmd/arcaflow/main.go

```diff
@@ -117,6 +123,19 @@ Options:
 	}
 
 	var err error
+
+	requiredFiles := map[string]string{
+		RequiredFileKeyConfig:   configFile,
+		RequiredFileKeyInput:    input,
+		RequiredFileKeyWorkflow: workflowFile,
+	}
+
+	requiredFilesAbsPaths, err := loadfile.ContextAbsFilepaths(dir, requiredFiles)
+	if err != nil {
+		tempLogger.Errorf("Failed to determine absolute path of arcaflow context directory %s (%v)", dir, err)
+		os.Exit(ExitCodeInvalidData)
+	}
```

Comment:
The lack of encapsulation here also bothers me: that is, any failure in `ContextAbsFilepaths` is assumed explicitly to be "failed to determine absolute path of [...] context directory", which seems an unnecessary external dependency on the function's internal behavior. What we actually know here is that it failed, not really why. And since you're reporting the `err` value as part of the new `error`, a message along the lines that "determining absolute context file paths failed" would be sufficient.

Reply:
The only error that can be thrown during `ContextAbsFilepaths` has to be raised from `filepath.Abs()`. On *nix systems, this leads to [unixAbs()](https://cs.opensource.google/go/go/+/refs/tags/go1.21.5:src/path/filepath/path.go;drc=9e933c189ca3a84f12995b3c799364a06abc4376;l=297) which itself is only going to raise an error if something particularly weird happens during the call to [os.Getwd()](https://cs.opensource.google/go/go/+/master:src/os/getwd.go;drc=5755bad42adc23ad4a0c32149ac8cf78ece5d0b0;l=22), so it seemed safe to make the error message that specific.

Reply:
I believe to consistently open and read the intended files, that we want to use the absolute file paths to the config, input, and workflow, so I think we need to do this first.

```
A -> B := B depends upon A

determine absolute file path -> read config file -> unmarshal config yaml into config struct -> instantiate main logger
```

Reply:
Fair enough:  I missed the detail that opening the config file depended on having the absolute path.

Does the ordering between the call to `flag.Usage()` and the logger matter?  (Because it's reversed between the `Abs()` error and the `LoadContext()` error.)  It might be handy to keep them consistent.

Reply:
> I missed the detail that opening the config file depended on having the absolute path.

Ah!  It _wasn't_ depending on it until your recent update!  😈